### DISCUSSION TITLE
Messages not back to the queues

### DISCUSSION THREAD

**Discussion by Sztacho** [php-amqplib/php-amqplib#1062] [Help wanted]
Hi, 
I do some staff in where I must use rabbitmq. I have defined queue and exchange. Exchange are binded with queue. 
 This is the config of exchange and queue.
![image](https://user-images.githubusercontent.com/22791103/212279147-b0506598-59d5-4d95-9217-ddb1a3b5dfac.png)

Now when I try use $message->nack() then message not back to the queue and this message are removed from queue. 

I use PHP 5.4, RabbitMq 3.11.4 and amqplib 2.12.3.

**Comment by lukebakken** [comment]
https://github.com/php-amqplib/php-amqplib/blob/master/PhpAmqpLib/Message/AMQPMessage.php#L113-L128

```
$message->nack(true, false)
```

**Reply by Sztacho** [reply]
Its work but this message now is back to the top of the queue not to the end. Some suggestion?

**Reply by Sztacho** [reply]
BUMP question.

**Reply by andrew-demb** [reply]
https://www.rabbitmq.com/nack.html

To move the message to the end you may need to publish it to the queue and ack current message.
